Library/Formula/flex.rb

```diff
@@ -15,6 +15,8 @@ class Flex < Formula
   depends_on "gettext"
 
   def install
+    # 2.5.38+ build fix adapted from http://patchwork.openembedded.org/patch/68029/
+    ENV["LDFLAGS"] = "#{ENV["LDFLAGS"]} -all-static"
```

Comment:
@tdsmith 
0. NFI, upstream broken their toolchain for osx and possibly other platforms on 2.5.38.  The patch was from some folks at WindRiver which is proposed as a fix on macports.
1. Nope.  It's not a good idea to have a shared version of `_yylex` floating around to break apps, probably should add a `--disable-shared` too, in which case, the libfl.la issue is moot and it's an artifact that's deleted by brew by default anyhow.  If someone absolutely needs a shared library, they'll append `-all-static` to `LDFLAGS`.

```
0 ~ ❯❯❯ nm -g /usr/local/Cellar/flex/2.5.39/lib/libfl.a

/usr/local/Cellar/flex/2.5.39/lib//libfl.a(libmain.o):
0000000000000000 T _main
0000000000000050 S _main.eh
                 U _yylex

/usr/local/Cellar/flex/2.5.39/lib//libfl.a(libyywrap.o):
0000000000000000 T _yywrap
0000000000000048 S _yywrap.eh
0 ~ ❯❯❯ 
```
2. Breaks libfl.la.

```
Undefined symbols for architecture x86_64:
  "_yylex", referenced from:
      _main in libmain.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libfl.la] Error 1
make[2]: *** Waiting for unfinished jobs....
Undefined symbols for architecture x86_64:
  "_yylex", referenced from:
      _main in libmain.o
```

3. Yup.